processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java

```diff
@@ -133,9 +138,16 @@ public int getPriority() {
         List<String> variableNames = new LinkedList<>();
 
         for ( TypeMirror usedMapper : mapperAnnotation.uses() ) {
+            boolean isSingleton = typeUtils.asElement( usedMapper ).getEnclosedElements().stream()
```

Comment:
I don't know this from the top of my head, but does a field in an interface automatically get the modifiers "PUBLIC, STATIC, FINAL" ? 

I guess checkstyle does, but should we warn as well if you spot an intention to use this pattern but for instance left out STATIC?

Reply:
Yes, a field on an interface is automatically "public static final".

Reply:
I don't think it is our job to emit warnings for this. If it is really needed every project can define different rules and have such warnings through checkstyle / sonar